src/app/_components/Search/Search.tsx

```diff
@@ -0,0 +1,450 @@
+              textTransform={'uppercase'}
+            >
+              {segment}
+            </Text>
+          ) : (
+            <React.Fragment key={index}>{segment}</React.Fragment>
+          )
+        )}
+    </Flex>
+  );
+}
+
+function SearchInput() {
+  const dispatch = useAppDispatch();
+  const tempSearchTerm = useAppSelector(selectTempSearchTerm);
+  const reg = new RegExp(`(${advancedSearchKeywords.join('|')})`, 'gi');
+  const isSearchFieldFocused = useAppSelector(selectIsSearchFieldFocused);
+  const router = useRouter();
+  const isSearchPage = usePathname() === '/search';
+  const network = useGlobalContext().activeNetwork;
+  const searchPageUrl = useSearchPageUrl(tempSearchTerm, network);
+  const isAdvancedSearch = advancedSearchKeywords.some(term => tempSearchTerm.includes(term));
+
+  const handleSearch = useCallback(() => {
+    if (isSearchPage && isAdvancedSearch) {
+      router.push(searchPageUrl);
+    } else {
+      dispatch(setSearchTerm(tempSearchTerm));
+    }
+  }, [dispatch, isAdvancedSearch, isSearchPage, router, searchPageUrl, tempSearchTerm]);
+
+  return (
+    <DoubleGradientBorderWrapper>
+      <Flex
+        background={'surfaceSecondary'}
+        alignItems={'center'}
+        width={'full'}
+        overflow="hidden"
+        position="relative"
+        borderRadius="redesign.lg"
+        zIndex={3}
+      >
+        <StartElement />
+        <Flex width={'full'} fontFamily="var(--font-instrument-sans)">
+          <HiddenSearchField
+            placeholder={isSearchFieldFocused ? '' : 'Explore'}
+            onKeyDown={(e: React.KeyboardEvent<HTMLInputElement>) => {
+              if (e.key === 'Enter') {
+                e.preventDefault();
+                handleSearch();
+              }
+            }}
+            onChange={(e: React.FormEvent<HTMLInputElement>) => {
+              const formattedValue = e.currentTarget.value.replace(reg, match => {
+                return match.toUpperCase();
+              });
+              dispatch(setTempSearchTerm(formattedValue.trimStart()));
+            }}
+            onScroll={(e: React.FormEvent<HTMLInputElement>) =>
+              dispatch(setSearchPreviewScrollLeft(e.currentTarget.scrollLeft))
+            }
+            value={tempSearchTerm}
+          />
+          <SearchPreview />
+        </Flex>
+        <EndElement handleSearch={handleSearch} />
+      </Flex>
+    </DoubleGradientBorderWrapper>
+  );
+}
+
+export function Search({ searchTermFromQueryParams = '' }: { searchTermFromQueryParams?: string }) {
+  const isSearchFieldFocused = useAppSelector(selectIsSearchFieldFocused);
+
+  return (
+    <HStack width="full" position={'relative'} zIndex={1}>
+      <SearchInput />
+      {isSearchFieldFocused && (
+        <SearchResultsWrapper>
+          <Stack gap={6}>
+            <Stack px={4} gap={3} pt={16}>
+              <KeywordsPreview />
+              <DocsLink />
+            </Stack>
+            <RecentResults />
```

Comment:
Shoudlnt this be put in the storybook and not in the actual component, as this will eventually become the real component that is used and will have live results